src/Common/CoTask.cpp

```diff
@@ -0,0 +1,168 @@
+#include <Common/CoTask.h>
+
+
+namespace DB::ErrorCodes
+{
+    extern const int TASK_CANCELLED;
+}
+
+
+namespace DB::Co
```

Comment:
This is too cool. Let's name it in a boring way: `Coroutine`.

Reply:
This prefix is used by C++ itself, see the keywords `co_await`, `co_return`.